### PR TITLE
[IMM32][SDK][NTUSER] Implement CtfImmTIMActivate

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose

To support Text Input Processors (TIPs).
JIRA issue: [CORE-19268](https://jira.reactos.org/browse/CORE-19268)

## Proposed changes

- Add `Imm32InitTLS`, `Imm32AllocateTLS`, `Imm32GetTLS`, `Imm32GetCoInitCountSkip`, `Imm32IncCoInitCountSkip`, and `Imm32DecCoInitCountSkip` helper functions to control the TLS data.
- Introduce `"CoInitialize Spy"` (`ISPY`) to manage COM initialization status.
- Implement `CtfImmCoInitialize` and `CtfImmCoUninitialize` functions.
- Implement `CtfImmEnterCoInitCountSkipMode` and `CtfImmLeaveCoInitCountSkipMode` functions.
- Implement `CtfImmLastEnabledWndDestroy`, `ImmDisableTextFrameService`, and `CtfImmTIMActivate` functions.

## TODO

- [x] Do build.

### CODE REVIEW

File: dll/win32/imm32/ctf.c [hunk 1]
```diff
@@ -203,6 +228,328 @@ Imm32CheckAndApplyAppCompat(
...
+    {
+        if (dwSkipCount)
+            pData->dwSkipCount = dwSkipCount - 1;
+    }
+
+    return dwSkipCount;
+}
+
+/***********************************************************************
+ * ISPY (I am not spy!)
```
Comment (HBelusca) [inline dll/win32/imm32/ctf.c:332]: What's its purpose?

Comment (katahiromz) [reply dll/win32/imm32/ctf.c:332]: COM-based Text Input Manager watches the `CoInitialize[Ex]`/`CoUninitialize` calls to get the timing of input activation/de-activation

Comment (katahiromz) [reply dll/win32/imm32/ctf.c:332]: Implementation is still incomplete.

**Comment by katahiromz** [comment]
Small tests are done. Let's go.

**Pull request merged**